src/de.rs

```diff
@@ -243,6 +243,48 @@ impl<'de, R: Read<'de>> Deserializer<R> {
         Error::syntax(reason, position.line, position.column)
     }
 
+    /// Returns the first non-comment byte without consuming it, or `None` if
+    /// EOF is encountered.
+    fn parse_comment(&mut self) -> Result<Option<u8>> {
+        // Cursor is already after the first slash.
+        match tri!(self.peek()) {
+            // Line Comment
+            Some(b'/') => {
+                self.eat_char();
+
+                // And eat the rest of the line.
+                loop {
+                    match tri!(self.peek()) {
```

Comment:
Does this work with a comment followed by EOF instead of a newline?

Reply:
It does! I'll get a test in to demonstrate that.